### ISSUE TITLE
cargo-dist `include` directive _may_ be having problems archiving subdirectories?

### ISSUE THREAD

**Issue by simonsan** [axodotdev/cargo-dist#873]
I'm actually not sure, where the problem lies. I opened this issue with scoop first, because it got apparent when I extracted a cargo-dist packaged app: https://github.com/ScoopInstaller/Scoop/issues/5845

But thinking about it, cargo-dist was zipping the file with the following `include` directive:

```toml
# Include files in the dist package
include = [
  "./config/",
  "./docs/",
  "./templates/",
]
```

So I wonder if there are some flags being used, that make it incompatible to be unpacked with 7-Zip?

A zip archive packed by cargo-dist like this:
```console
│   CHANGELOG.md
│   LICENSE
│   pace.exe
│   README.md
│
├───config
│       pace.toml
│       projects.pace.toml
│       README.md
│       tasks.pace.toml
│
├───docs
│       LICENSE
│       pace-dev-docs.pdf
│       pace-user-docs.pdf
│       README.md
│
└───templates
    │   pace_report_json.html
    │   README.md
    │
    └───reflections
            basic.html
            basic.md
```

is being extracted like this on Win11-x64 with 7-Zip: 23.01-x64

```console
│   CHANGELOG.md
│   configpace.toml
│   configprojects.pace.toml
│   configREADME.md
│   configtasks.pace.toml
│   docsLICENSE
│   docspace-dev-docs.pdf
│   docspace-user-docs.pdf
│   docsREADME.md
│   install.json
│   LICENSE
│   manifest.json
│   pace.exe
│   README.md
│   templatespace_report_json.html
│   templatesREADME.md
│   templatesreflectionsbasic.html
│   templatesreflectionsbasic.md
│
├───config                       # persisted from before
│       pace.toml                # persisted from before
│       pace.toml.bak            # persisted from before
│
├───config.original
├───docs                         # persisted from before
├───logs                         # persisted from before
├───templates
└───templatesreflections
```

I'm talking about this archive: https://github.com/pace-rs/pace/releases/download/pace-rs-v0.15.1/pace-rs-x86_64-pc-windows-msvc.zip

I tried to get to the ground of it, but got lost in the Github Actions file. If you could direct to the root of this, I would be able to investigate further. Thank you!

**Comment by mistydemeo** [comment]
Wow, that's weird. I'm seeing the same thing with `unar` and `unzip -l`; the forward slash is part of the filenames instead of the proper directory structure being included. I'll look into it.

**Comment by mistydemeo** [comment]
It looks like the tarballs are fine; only the zips are affected. I'm looking into the generation to identify the issue.

**Comment by simonsan** [comment]
Oh wow, thanks. I wanted to confirm, first, what's going on. Thank you a lot for looking into it. <3

**Comment by sorokya** [comment]
Will this fix be released soon? My users are mostly on windows and the broken zip is pretty annoying.

Thanks

**Comment by mistydemeo** [comment]
It's a holiday weekend at the moment, so I've been out of the office a little longer than usual. The next release will most likely be this coming week.

**Comment by sorokya** [comment]
> It's a holiday weekend at the moment, so I've been out of the office a little longer than usual. The next release will most likely be this coming week.

No rush! Just wanted to confirm. Thanks. Happy Easter 😊

**Comment by mistydemeo** [comment]
This fix is now available in the 0.12.1 bugfix release.

**Comment by simonsan** [comment]
Thanks a lot!

**Comment by sorokya** [comment]
I think there might still be an issue with sub directories.

In my project I'm trying to include `pub/items` but in the zip I get a folder named `pub` (with the readme file inside as expected! This was fixed)

But instead of a subdirectory in `pub` for `items` I get a folder in the root of the zip named `pub\items`

Can look at the latest release of my repo to see the issue.

[reoserv](https://github.com/sorokya/reoserv)

**Comment by mistydemeo** [comment]
I'll test with that repo and see if I can identify the issue. Sorry about the inconvenience!

**Comment by mistydemeo** [comment]
I believe this should be fixed with the just-released 0.13.2; I tested with reoserv on Windows and the ZIP structure looks good to me now. Can you confirm if this is working for you now?

**Comment by sorokya** [comment]
Thanks! I'll test this soon

**Comment by sorokya** [comment]
> I believe this should be fixed with the just-released 0.13.2; I tested with reoserv on Windows and the ZIP structure looks good to me now. Can you confirm if this is working for you now?

This is working now for me! thanks so much. Can close issue

**Comment by mistydemeo** [comment]
You're welcome! Glad it's working for you.
